integration-tests/wrappers/contract_caller.go

```diff
@@ -0,0 +1,177 @@
+	client := w.getGethClient()
+	return client.CallContractAtHash(ctx, call, blockHash)
+}
+
+func (w *WrappedContractBackend) HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error) {
+	client := w.getGethClient()
+	return client.HeaderByNumber(ctx, number)
+}
+
+func (w *WrappedContractBackend) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
+	client := w.getGethClient()
+	return client.PendingNonceAt(ctx, account)
+}
+
+func (w *WrappedContractBackend) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
+	client := w.getGethClient()
+	return client.SuggestGasPrice(ctx)
+}
+
+func (w *WrappedContractBackend) SuggestGasTipCap(ctx context.Context) (*big.Int, error) {
+	client := w.getGethClient()
+	return client.SuggestGasTipCap(ctx)
+}
+
+func (w *WrappedContractBackend) EstimateGas(ctx context.Context, call ethereum.CallMsg) (gas uint64, err error) {
+	client := w.getGethClient()
+	return client.EstimateGas(ctx, call)
+}
+
+func (w *WrappedContractBackend) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	client := w.getGethClient()
+	return client.SendTransaction(ctx, tx)
+}
+
+func (w *WrappedContractBackend) FilterLogs(ctx context.Context, query ethereum.FilterQuery) ([]types.Log, error) {
+	client := w.getGethClient()
+	return client.FilterLogs(ctx, query)
+}
+
+func (w *WrappedContractBackend) SubscribeFilterLogs(ctx context.Context, query ethereum.FilterQuery, ch chan<- types.Log) (ethereum.Subscription, error) {
+	client := w.getGethClient()
+	return client.SubscribeFilterLogs(ctx, query, ch)
+}
+
+func (w *WrappedContractBackend) CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error) {
+	var hex hexutil.Bytes
+	client := w.getGethClient()
+	err := client.Client().CallContext(ctx, &hex, "eth_call", toCallArg(msg), toBlockNumArg(blockNumber))
+	if err != nil {
+		return nil, err
+	}
+	return hex, nil
+}
+
+func (w *WrappedContractBackend) PendingCallContract(ctx context.Context, msg ethereum.CallMsg) ([]byte, error) {
+	var hex hexutil.Bytes
+	client := w.getGethClient()
+	err := client.Client().CallContext(ctx, &hex, "eth_call", toCallArg(msg), "pending")
+	if err != nil {
+		return nil, err
+	}
+	return hex, nil
+}
+
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
+func toBlockNumArg(number *big.Int) string {
+	if number == nil {
+		return "latest"
+	}
+	if number.Sign() >= 0 {
+		return hexutil.EncodeBig(number)
+	}
+	// It's negative.
+	if number.IsInt64() {
+		return rpc.BlockNumber(number.Int64()).String()
+	}
+	// It's negative and large, which is invalid.
+	return fmt.Sprintf("<invalid %d>", number)
+}
+
+// COPIED FROM go-ethereum/ethclient/gethclient - must be kept up to date!
+// Modified to include legacy 'data' as well as 'input' in order to support non-compliant servers.
+func toCallArg(msg ethereum.CallMsg) interface{} {
```

Comment:
I'm assuming there's no chance we can convince them to make this a public function? Is there anyway we can make checking for this part of the GitHub action? We will definitely forget.

Reply:
EXPORTED!